core/templates/pages/topic-editor-page/services/topic-editor-routing.service.ts

```diff
@@ -15,102 +15,112 @@
       } else if (newPath.startsWith('/subtopic_editor')) {
-        activeTabName = SUBTOPIC_EDITOR_TAB;
+        this.activeTabName = this.SUBTOPIC_EDITOR_TAB;
       } else if (newPath.startsWith('/subtopic_preview')) {
-        activeTabName = SUBTOPIC_PREVIEW_TAB;
+        this.activeTabName = this.SUBTOPIC_PREVIEW_TAB;
       } else if (newPath.startsWith('/topic_preview')) {
-        activeTabName = TOPIC_PREVIEW_TAB;
+        this.activeTabName = this.TOPIC_PREVIEW_TAB;
       }
     });
+    this.location.go(this.currentPath);
+  }
 
-    var TopicEditorRouterService = {
-      getActiveTabName: function() {
-        return activeTabName;
-      },
-      getLastTabVisited: function() {
-        return lastTabVisited;
-      },
-      getLastSubtopicIdVisited: function() {
-        return lastSubtopicId;
-      },
-      navigateToMainTab: function() {
-        lastTabVisited = 'topic';
-        PageTitleService.setPageTitleForMobileView('Topic Editor');
-        $location.path('');
-      },
-      navigateToSubtopicPreviewTab: function(subtopicId) {
-        lastTabVisited = 'subtopic';
-        PageTitleService.setPageTitleForMobileView('Subtopic Preview');
-        $location.path('/subtopic_preview/' + subtopicId);
-      },
-      navigateToTopicPreviewTab: function() {
-        lastTabVisited = 'topic';
-        PageTitleService.setPageTitleForMobileView('Topic Preview');
-        $location.path('/topic_preview/');
-      },
-      navigateToSubtopicEditorWithId: function(subtopicId) {
-        lastTabVisited = 'subtopic';
-        PageTitleService.setPageTitleForMobileView('Subtopic Editor');
-        $location.path('/subtopic_editor/' + subtopicId);
-      },
-      navigateToQuestionsTab: function() {
-        lastSubtopicId = this.getSubtopicIdFromUrl();
-        PageTitleService.setPageTitleForMobileView('Question Editor');
-        $location.path('/questions');
-      },
-      getSubtopicIdFromUrl: function() {
-        return parseInt($location.path().split('/')[2]);
-      },
-      navigateToSkillEditorWithId: function(skillId) {
-        var SKILL_EDITOR_URL_TEMPLATE = '/skill_editor/<skill_id>';
-
-        var skillEditorUrl = UrlInterpolationService.interpolateUrl(
-          SKILL_EDITOR_URL_TEMPLATE, {
-            skill_id: skillId
-          });
-        $window.open(skillEditorUrl);
-      }
-    };
+  getActiveTabName(): string {
+    return this.activeTabName;
+  }
+
+  getLastTabVisited(): string {
+    return this.lastTabVisited;
+  }
 
-    return TopicEditorRouterService;
+  getLastSubtopicIdVisited(): number {
+    return this.lastSubtopicId;
   }
-]);
+
+  navigateToMainTab(): void {
+    this.lastTabVisited = 'topic';
+    this.location.go('#/');
+  }
+
+  navigateToSubtopicPreviewTab(subtopicId: string | number): void {
+    this.lastTabVisited = 'subtopic';
+    this.pageTitleService.setPageTitleForMobileView('Subtopic Preview');
+    this.location.go('#/subtopic_preview/' + subtopicId);
```

Comment:
Did you check these changes?

Reply:
Yes I did there was no problem with the frontend tests and the pre-push tests also went through

Reply:
No, by starting the server and checking if the routing is working properly or not.

Reply:
Okay Im checking that